ros2bag/ros2bag/verb/record.py

```diff
@@ -80,6 +104,32 @@ def create_bag_directory(self, uri):
         except OSError:
             return "[ERROR] [ros2bag]: Could not create bag folder '{}'.".format(uri)
 
+    def _dict_to_duration(self, time_dict: Optional[Dict[str, int]]) -> Duration:
+        if not all(key in time_dict for key in ['sec', 'nsec']):
```

Comment:
The logic in this function is really confusing. Let's remove all those negations:

```python

  if time_dict:
    if all(...):
       return ...
    else:
      raise ...
  else:
    return Duration()
```

Reply:
Additionally, consider moving your `all(...)` into another function to make the logic clearer.

`def _is_dict_valid_duration(dict) -> bool`

Reply:
It was a previous suggestion to fail early than later, which is why we have that negation statement upfront.

Reply:
How is this making a difference?